src/S3.ts

```diff
@@ -16,43 +17,51 @@ class S3 {
   }
 
   public async initialise() {
-    const s3UrlBase: any = new S3File.Endpoint(this.url)
-    this.s3Handler = new S3File.S3({
-      endpoint: s3UrlBase,
-      accessKeyId: this.params.keyId,
-      secretAccessKey: this.params.secretAccessKey,
-      s3ForcePathStyle: s3UrlBase.protocol === 'http:',
+    const s3UrlObj: any = new URL(this.url)
+    const regionRegex: RegExp = /^https?:\/\/s3\.([^.]+)\.amazonaws\.com/;
+    const match: RegExpMatchArray | null = this.url.match(regionRegex);
+    if (match && match[1]) {
+       this.region = match[1];
+    }
+    this.s3Handler = new S3Client({
+      region: this.region,
+      credentials: {
+        accessKeyId: this.params.keyId,
+        secretAccessKey: this.params.secretAccessKey,
+      },
+      endpoint: s3UrlObj.href,
+      forcePathStyle: s3UrlObj.protocol === 'http:',
     })
-    try {
-      if ((await this.bucketExists(this.bucketName)) === true) {
-        return true
-      }
-    } catch (err) {
-      throw new Error(`Unable to connect to S3, either S3 login credentials are wrong or bucket cannot be found
```

Comment:
This error is not explicit enough, probably **err.message** should be used instead.